src/web-components.js

```diff
@@ -48,6 +62,29 @@ export function isShadowDomSupported() {
 }
 
 /**
+ * Returns `true` if Shadow CSS encapsulation is supported.
+ * @return {boolean}
+ */
+export function isShadowCssSupported() {
+  if (disableShadowCss === undefined) {
```

Comment:
This is a bit problematic b/c we force the slowest path on the most likely scenario. Let's redo a bit differently. Similar to `shadowDomSupported` above. It'd look something like this:

```
/**
 * @type {boolean|undefined}
 */
let shadowCssSupported;

export function setShadowCssSupportedForTesting(value) {
  shadowCssSupported = value;
}

/**
 * @return {boolean}
 */
export function isShadowCssSupported() {
  if (shadowCssSupported === undefined) {
    shadowCssSupported = isShadowDomSupported() && (... || ...);
  }
  return shadowCssSupported;
}
```

This why the check will be done only once and then will work reliably and fast.

Reply:
Awesome, I'll get this done later today :)

Reply:
Done. My apologies on the late reply, had some school stuff :)